1.5 py3.11 builds failing to build podbc

`     gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -Isasl -I/opt/hostedtoolcache/Python/3.11.4/x64/include/python3.11 -c sasl/saslwrapper.cpp -o build/temp.linux-x86_64-cpython-311/sasl/saslwrapper.o
      sasl/saslwrapper.cpp:196:12: fatal error: longintrepr.h: No such file or directory
        196 |   #include "longintrepr.h"
            |            ^~~~~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
...

Failed to build pyodbc sasl
ERROR: Could not build wheels for pyodbc, sasl, which is required to install pyproject.toml-based projects
`
https://github.com/dbt-labs/dbt-core-snapshots/actions/runs/5326716584/jobs/9649283636#step:7:1028

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -29,6 +29,7 @@ jobs:
       fail-fast: false
       matrix:
        os: ["macos-latest", "ubuntu-latest"]
+       python-version: ["3.8", "3.11"]
     runs-on: ${{ matrix.os }}
     name: ${{ matrix.os }} Test
 
@@ -49,7 +50,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: "3.8"
+          python-version: ${{ matrix.python-version }}
 
       - name: Set up Linux
         if: matrix.os == 'ubuntu-latest'

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -31,7 +31,7 @@ jobs:
        os: ["macos-latest", "ubuntu-latest"]
        python-version: ["3.8", "3.11"]
     runs-on: ${{ matrix.os }}
-    name: ${{ matrix.os }} Test
+    name: ${{ matrix.os }} ${{ matrix.python-version }} Test
 
     steps:
       - name: Check out the repository (non-PR)

--- a/release_creation/bundle/requirements/v1.5.latest.requirements.txt
+++ b/release_creation/bundle/requirements/v1.5.latest.requirements.txt
@@ -10,5 +10,5 @@ dbt-rpc~=0.4.0
 grpcio-status~=1.47.0
 pyarrow~=12.0.0,!=12.0.1
 pyasn1-modules~=0.2.1
-pyodbc==4.0.32 --no-binary pyodbc
+pyodbc==4.0.39 --no-binary pyodbc
 snowflake-connector-python~=3.0,!=3.0.4

--- a/release_creation/bundle/requirements/v1.5.pre.requirements.txt
+++ b/release_creation/bundle/requirements/v1.5.pre.requirements.txt
@@ -9,5 +9,5 @@ dbt-rpc~=0.1.3b1
 grpcio-status~=1.47.0
 pyarrow~=12.0.0,!=12.0.1
 pyasn1-modules~=0.2.1
-pyodbc==4.0.32 --no-binary pyodbc
+pyodbc==4.0.39 --no-binary pyodbc
 snowflake-connector-python~=3.0,!=3.0.4

--- a/release_creation/bundle/requirements/v1.6.pre.requirements.txt
+++ b/release_creation/bundle/requirements/v1.6.pre.requirements.txt
@@ -9,5 +9,5 @@ dbt-rpc~=0.4.1
 grpcio-status~=1.47.0
 pyasn1-modules~=0.2.1
 pyarrow~=12.0.0,!=12.0.1
-pyodbc==4.0.32 --no-binary pyodbc
+pyodbc==4.0.39 --no-binary pyodbc
 snowflake-connector-python~=3.0,!=3.0.4

--- a/test/integration/bundle/test_create.py
+++ b/test/integration/bundle/test_create.py
@@ -9,7 +9,7 @@ import zipfile
 
 
 @pytest.mark.parametrize(argnames="test_version",
-                         argvalues=["1.1.0", "1.2.0", "1.3.0", "1.4.0", "1.5.0", "1.6.0b2"])
+                         argvalues=["1.3.0", "1.4.0", "1.5.0", "1.6.0b2"])
 def test_generate_bundle_creates_a_bundle_with_valid_version(test_version):
     created_assets = generate_bundle(Version.coerce(test_version))
     for asset_name, asset_location in created_assets.items():

